apps/communications/contacts/js/contacts.js

```diff
@@ -484,18 +486,52 @@ var Contacts = (function() {
     }
   }
 
-  var goToSelectTag = function goToSelectTag(event) {
-    contactTag = event.currentTarget.children[0];
+  function showDetails() {
+    navigation.go('add-details-view', 'right-left');
+    if (!tagAddDetails) {
+      tagAddDetails = document.querySelector('#add-details-header');
+      tagAddDetails.addEventListener('action', handleBack);
+    }
+    var boxTitle = document.querySelectorAll('li.action');
+    for(var i=0; i<boxTitle.length; i++) {
+      if (!boxTitleTags[i]) {
+        boxTitle[i].addEventListener('click', goToSelectTag);
+        boxTitleTags[i] = true;
+      }
+    }
+    if (document.activeElement) {
+      document.activeElement.blur();
+    }
+  }
 
-    var tagViewElement = document.getElementById('view-select-tag');
-    if (!lazyLoadedTagsDom) {
+  var goToAddDetails = function goToAddDetails(event) {
+    var tagViewElement = document.getElementById('add-details-view');
+    if (!lazyLoadedDetailsDom) {
       LazyLoader.load(tagViewElement, function() {
-        showSelectTag();
-        lazyLoadedTagsDom = true;
-       });
+        lazyLoadedDetailsDom = true;
+        showDetails();
+      });
     }
     else {
-      showSelectTag();
+        showDetails();
+    }
+  };
+
+  var goToSelectTag = function goToSelectTag(event) {
+    contactTag = event.currentTarget.children[0];
+    if(contactTag.textContent==='Company'){
+      showComapny();
```

Comment:
>> Vinay Mishra 
>>I am using this function “showCompany”, bcz of more clear and reusable point.  If you want then I will remove it. 

OK, let it be here, it would be better to update usage of 'companyNameHidden' class